web/blog/tenzir-for-splunk-users/index.md

```diff
@@ -0,0 +1,320 @@
+---
+title: Tenzir for Splunk Users
+authors: mavam
+date: 2023-08-03
+tags: [zeek, threat hunting, pipelines, tql, splunk, spl]
+comments: true
+---
+
+The syntax of [Tenzir Query Language (TQL)](/language) looks remarkably similar
+to Splunk's [Search Processing Language (SPL)][spl]. This is by no means an
+accident. Having spoken with numerous security practitioners, we found a
+universally positive sentiment about the ease of use. In this blog post, we
+explain how the two languages differ using threat hunting examples.
+
+[spl]: https://docs.splunk.com/Documentation/SplunkCloud/latest/Search/Aboutthesearchlanguage
+
+![SPL versus TQL](spl-vs-tql.excalidraw.svg)
+
+<!--truncate-->
+
+## Why not SQL?
+
+Splunk was the first tool that provided an integrated solution from interactive
+data exploration to management-grade dashboards—all powered by dataflow
+pipelines. The success of Splunk is not only resulting from their first-mover
+advantage in the market, but also because their likable user experience: it is
+*easy* to get things done.
+
+At Tenzir, we have a very clear target audience: security practitioners. They
+are not necessarily data engineers and fluent in SQL and low-level data tools,
+but rather identify as blue teamers, incident responders, threat hunters,
+detection engineers, threat intelligence analysts, and other domain experts. Our
+goal is cater to these folks, without requiring them to have deep understanding
+of relational algebra.
+
+We opted for a dataflow language because it simplifies reasoning—one step at a
+time. At least conceptually, because a smart system optimizes the execution
+under the hood. Any declarative language where users describe the desired
+outcome offers the exeuction engine to re-arrange the input, as long as it
+doesn't change the output. Particularly SQL has property, where relational
+algebra describes the *what* and not the *how*. A dataflow language is a bit
```

Comment:
Alternatively, here is a rough rewording suggestion:
```suggestion
under the hood. As long as the observable behavior remains the same, the underlying 
implementation can optimize the actual computation at will. This is especially 
noticable with declarative languages such as SQL, where the user describes the 
*what* instead of the *how*. A dataflow language is a bit
```

Reply:
59f90b29d4